src/screens/PostsDisplayScreen/PostCard/PostCard.tsx

```diff
@@ -0,0 +1,374 @@
+import {useDispatch, useSelector} from "react-redux";
+import React, {useEffect, useState} from "react";
+import axios from "axios";
+import {URI} from "../../../URI";
+import {TouchableOpacity, View, Image, TouchableWithoutFeedback, Text, Modal, StyleSheet} from "react-native";
+import {addLikes, removeLikes} from "../../../redux/actions/postActions";
+import PostDetailsCard from "./PostDetailsCard";
+import getTimeDuration from "./timeGen";
+import {useNavigation} from "@react-navigation/native";
+import {Post, User} from "../../../redux/reducers/User";
+import {getPosts} from "../../../redux/actions/getPosts";
+
+type Props = {
+    item: Post;
+    isReply?: boolean | null;
+    postId?: string | null;
+    replies?: boolean | null;
+};
```

Comment:
I'm a little bit confused with this. Why do we need the optional fields here? Doesn't `Post` has an `id` and we need a `postId`? If `replies` is a boolean, we can use `hasReplies` to make things easier